data_structures/arrays/sprialmatrix.py

```diff
@@ -0,0 +1,54 @@
+def spiralorder(matrix: list[list[int]]) -> list[int]:
+    """
+    Get the elements of a 2D matrix in spiral order.
+
+    Args:
+    matrix (list[list[int]]): The input 2D matrix.
+
+    Returns:
+    List[int]: A list containing the elements of the matrix in spiral order.
+
```

Comment:
______________________ [doctest] sprialmatrix.spiralorder ______________________
004 
005     Args:
006     matrix (list[list[int]]): The input 2D matrix.
007 
008     Returns:
009     List[int]: A list containing the elements of the matrix in spiral order.
010 
011     Example:
012     >>> a = [[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12], [13, 14, 15, 16]]
013     >>> spiralOrder(a)